inference-engine/src/multi_device/multi_device_plugin.cpp

```diff
@@ -163,13 +164,16 @@ ExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadExeNetworkImpl(co
     // collect the settings that are applicable to the devices we are loading the network to
     std::unordered_map<std::string, InferenceEngine::Parameter> multiNetworkConfig;
     multiNetworkConfig.insert(*priorities);
-
+    if (network.getFunction() == nullptr) {
+        THROW_IE_EXCEPTION << "MULTI device supports just ngraph network representation";
+    }
     DeviceMap<ExecutableNetwork> executableNetworkPerDevice;
     for (auto& p : metaDevices) {
         auto & deviceName = p.deviceName;
         auto & deviceConfig = p.config;
-        auto clonedNetwork = cloneNetwork(network);
-        executableNetworkPerDevice.insert({ deviceName, GetCore()->LoadNetwork(CNNNetwork{clonedNetwork}, deviceName, deviceConfig) });
+        executableNetworkPerDevice.insert({ deviceName, GetCore()->LoadNetwork(
+            CNNNetwork{ICNNNetwork::Ptr{const_cast<ICNNNetwork*>(&network),
```

Comment:
It looks like a hack, that we take an address and put it to shared pointer.
@ilya-lavrenov maybe we need to think about some changes in plugin API and use the CNNNetwork instead of ICNNNetwork.

Reply:
> maybe we need to think about some changes in plugin API and use the CNNNetwork instead of ICNNNetwork.

I agree